File: fix_bom.js
const fs = require('fs');
const path = require('path');

function removeBomFromDir(dir) {
    const files = fs.readdirSync(dir);

    files.forEach(file => {
        const filePath = path.join(dir, file);
        const stats = fs.statSync(filePath);

        if (stats.isDirectory()) {
            removeBomFromDir(filePath);
        } else if (filePath.endsWith('.java')) {
            const buffer = fs.readFileSync(filePath);
            if (buffer.length >= 3 && buffer[0] === 0xEF && buffer[1] === 0xBB && buffer[2] === 0xBF) {
                console.log(`Removing BOM from: ${filePath}`);
                const newBuffer = buffer.slice(3);
                fs.writeFileSync(filePath, newBuffer);
            }
        }
    });
}

const targetDir = String.raw`c:\Users\Oguzhan\Desktop\Repositories\ApplyFollow\backend\src`;
console.log(`Scanning ${targetDir}...`);
removeBomFromDir(targetDir);
console.log('Done.');

File: fix_bom.py
import os

def remove_bom(base_path):
    print(f"Scanning for files with BOM in {base_path}...")
    count = 0
    for root, dirs, files in os.walk(base_path):
        for file in files:
            if file.endswith(".java"):
                file_path = os.path.join(root, file)
                try:
                    with open(file_path, 'rb') as f:
                        content = f.read()
                    
                    if content.startswith(b'\xef\xbb\xbf'):
                        print(f"Removing BOM from: {file}")
                        new_content = content[3:]
                        with open(file_path, 'wb') as f:
                            f.write(new_content)
                        count += 1
                except Exception as e:
                    print(f"Error processing {file_path}: {e}")

    print(f"Fixed encoding for {count} files.")

if __name__ == "__main__":
    remove_bom(r"c:\Users\Oguzhan\Desktop\Repositories\ApplyFollow\backend\src")
